### ISSUE TITLE
Did you forget to write "loss_model.eval()"?

### ISSUE THREAD

**Issue by yuziGuo** [UKPLab/sentence-transformers#162]
```
    def _eval_during_training(self, evaluator, output_path, save_best_model, epoch, steps):
        """Runs evaluation during the training"""
        if evaluator is not None:
            score = evaluator(self, output_path=output_path, epoch=epoch, steps=steps)
            if score > self.best_score and save_best_model:
                self.save(output_path)
                self.best_score = score
```

The method is called here:
```
                if evaluation_steps > 0 and training_steps % evaluation_steps == 0:
                    self._eval_during_training(evaluator, output_path, save_best_model, epoch, training_steps)
                    for loss_model in loss_models:
                        loss_model.zero_grad()
                        loss_model.train()
```

**Comment by nreimers** [comment]
Hi @yuziGuo 
Thank you for checking this.

The call for model.eval() is in the evaluators, e.g.
https://github.com/UKPLab/sentence-transformers/blob/master/sentence_transformers/evaluation/LabelAccuracyEvaluator.py

```
def __call__(self, model, output_path: str = None, epoch: int = -1, steps: int = -1) -> float:
        model.eval()
        total = 0
        correct = 0
```

**Comment by wadhwasahil** [comment]
I think @yuziGuo is right,
when we call LabelAccuracyEvaluator, there should be a call like `self.softmax_model.eval()` because softmax_model is also `nn.module` type.

@nreimers what do you say?

**Comment by nreimers** [comment]
Hi @wadhwasahil 
Yes, there could be an softmax_model.eval() be added.

However, it will not make any difference. eval() disables dropout / batch-norm layers. However, softmax_model does not use dropout or batch-normalization, i.e., the results will be the same.

But it would be cleaner to call there also softmax_model.eval()
